src/rotten-tomatoes/index.js

```diff
@@ -1,3 +1,80 @@
-function rottenTomatoes(grid) {}
+function rottenTomatoes(grid) {
+    let gridCopy = grid
+
+  // An array that holds added values to the four adjecent nodes
+  // top, bottom, left and right respectively.
+    const dimensions = [
+      [-1, 0],
+      [1, 0],
+      [0, -1],
+      [0, 1]
+    ];
+
+    let row = 0;
+    let col = 0;
+    let completed = false;
+    let queue = [];
+    let steps = 0;
+
+  // A while loop that keeps running-
+  // so far as there are still possible adjacent fresh tomato
+    while (!completed) {
+
+      // Looping through the grid, seacrching for rotten tomatos-
+      // that has adjacent fresh tomatoes
+      gridCopy.forEach((element, rowI) => {
+          element.forEach((item, colI) => {
+
+            // If the present tomato is rotten,
+            // it checks to see if it'll find adjacent fresh tomato
+            // and pushes its postion into the queue
+              if (item == 2) {
+
+                  dimensions.forEach(position => {
+
+                      row = rowI + position[0];
+                      col = colI + position[1];
+
+                    // Avoiding undefind values(out of boundry positions)
+                    if (!(row < 0) && !(col < 0) &&
+                      !(row >= gridCopy.length) &&
+                      !(col >= gridCopy[0].length)) {
+
+                      // If the tomato at the current position is fresh,
+                      // its position is then pushed into the queue
+                      if (gridCopy[row][col] == 1) {
+                        queue.push([row, col])
+                      }
+                    }
+
+                  })
+              }
+          })
+      });
+
+      // Checks if queue is empty; meaninig that no other adjacent fresh tomato was found
+        if (queue.length == 0) {
+            completed = true
+        } else {
+          // If the quene is not empty, then there are still more possible fresh tomatos
+          steps++
+          // Then they're made to rot one after another
+          queue.forEach(item => gridCopy[item[0]][item[1]] = 2);
```

Comment:
I see you mutate the grid here. With your current way of copying, the original `grid` is still mutated.

Why does this need to happen though?

Reply:
I thought it's the only way to change the 1s on the grid to 2.
Or, is there any other way I can do it better?